doc/rbd/rbd-snapshot.rst

```diff
@@ -5,26 +5,29 @@
 .. index:: Ceph Block Device; snapshots
 
 A snapshot is a read-only logical copy of an image at a particular point in
-time: a checkpoint. One of the advanced features of Ceph block devices is
-that you can create snapshots of images to retain point-in-time state history.
-Ceph also supports snapshot layering, which allows you to clone images (e.g., a
-VM image) quickly and easily. Ceph block device snapshots are managed using the
-``rbd``  command and multiple higher level interfaces, including `QEMU`_,
-`libvirt`_, `OpenStack`_ and `CloudStack`_.
+time: a checkpoint. One of the advanced features of Ceph block devices is that
+you can create snapshots of images to retain point-in-time state history.  Ceph
+also supports snapshot layering, which allows you to clone images (for example,
+VM images) quickly and easily. Ceph block device snapshots are managed using
+the ``rbd`` command and several higher-level interfaces, including `QEMU`_,
+`libvirt`_, `OpenStack`_, and `CloudStack`_.
 
 .. important:: To use RBD snapshots, you must have a running Ceph cluster.
 
-.. note:: Because RBD does not know about any filesystem within an image
-          (volume), snapshots are only `crash-consistent` unless they are
-          coordinated within the mounting (attaching) operating system.
-          We therefore recommend that you pause or stop I/O before taking a snapshot.
-          If the volume contains a filesystem, it should be in an internally
-          consistent state before taking a snapshot.  Snapshots taken without
-          write quiescing may need an `fsck` pass before subsequent
-          mounting.  To quiesce `I/O` you can use `fsfreeze` command. See
-	  `fsfreeze(8)` man page for more details.
-	  For virtual machines, `qemu-guest-agent` can be used to automatically
-	  freeze file systems when creating a snapshot.
+
+.. note:: Because RBD is unaware of any file system within an image (volume),
```

Comment:
RBD is unaware of the contents of the image and the "thing" that is using the image: it could be a filesystem (potentially a root filesystem if RBD is backing a VM), someone using a block device directly, etc.  The notion of consistency is specific to that "thing".  A basic crash consistency, filesystem-level consistency and application-level consistency are distinguished (at least).

> Does this mean to say that snapshots are crash-consistent only if they are not co-ordinated within the mounting operating system?

No, RBD snapshots should always be crash-consistent.

> Also, what does it mean to be co-ordinated within that operating system?

It means letting the operating system know that the snapshot is about to be taken so that it can take any actions necessary to ensure required consistency (notify applications, quiesce filesystems, etc).

Reply:
I think this statement is correct as is:
- by default (no coordination), snapshots are crash-consistent
- depending on the level of coordination, snapshot can become filesystem-level or application-level consistent

Crash consistency is the weakest level of consistency, hence the use of "only".

Reply:
Sure!

Reply:
OK, thanks!